src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,439 @@
+          readonly={this.readonly}
+          ref={this.setTextareaEl}
+          required={this.required}
+          rows={this.rows}
+          value={this.value}
+          wrap={this.wrap}
+        />
+        <span class="content">
+          <slot onSlotchange={this.contentSlotChangeHandler} />
+        </span>
+        {
+          <footer
+            class={{
+              [CSS.footer]: true,
+              [CSS.readonly]: this.readonly,
+              [CSS.hide]: !hasFooter
+            }}
+            key={CSS.footer}
+            ref={(el) => (this.footerEl = el as HTMLElement)}
+          >
+            <slot name={SLOTS.footerStart} onSlotchange={this.footerTrailingSlotChangeHandler} />
+            <slot name={SLOTS.footerEnd} onSlotchange={this.footerLeadingSlotChangeHandler} />
+            {this.renderCharacterLimit()}
+          </footer>
+        }
+        <HiddenFormInputSlot component={this} />
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Sets focus on the component. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    this.textareaEl.focus();
+  }
+
+  /** Selects all text of the component's `value`. */
+  @Method()
+  async selectText(): Promise<void> {
+    this.textareaEl.select();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  formEl: HTMLFormElement;
+
+  defaultValue: Textarea["value"];
+
+  labelEl: HTMLCalciteLabelElement;
+
+  textareaEl: HTMLTextAreaElement;
+
+  footerEl: HTMLElement;
+
+  @State() effectiveLocale = "";
+
+  @State() trailingSlotElements: Element[];
+
+  @State() leadingSlotElements: Element[];
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  @State() defaultMessages: TextareaMessages;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  onFormReset(): void {
+    this.value = this.defaultValue;
+  }
+
+  onLabelClick(): void {
+    this.setFocus();
+  }
+
+  handleInput = (event: InputEvent): void => {
+    this.value = event.target["value"];
+    this.calciteTextareaInput.emit();
+  };
+
+  handleChange = (): void => {
+    this.calciteTextareaChange.emit();
+  };
+
+  footerTrailingSlotChangeHandler = (event: Event): void => {
+    this.trailingSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.trailingSlotElements);
+    }
+  };
+
+  footerLeadingSlotChangeHandler = (event: Event): void => {
```

Comment:
Nitpick: should we change these handlers to Start/End instead of Leading/Trailing?